### PR TITLE
feat: Remove quote requests from cart

### LINKED ISSUE #948: After adding a quote to cart these items cannot be deleted from cart
**Actual Behavior**
After adding a quote to cart the items cannot be deleted.

**Expected Behavior**
There is a delete button to delete cart items that are added from a quote.

**Steps to Reproduce the Bug**

1. Login as a b2b customer
2. Create a quote request and submit it
3. In the ICM backoffice create a quote response
4. Add your quote to the basket (see #947)
5. Go to cart: There is no delete button to remove these items, the user has to order them

**Environment Details**

- Desktop: \*

  - Browser [e.g. chrome, safari], Version [e.g. 22]
  - Project Code Version

- Smartphone: \*
  - Browser [e.g. stock browser, safari], Version [e.g. 22]
  - Project Code Version

**Additional Context, like Screenshots, Log File Snippets etc.**

[AB#71909](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/71909)

### PR DESCRIPTION
**Pull request by luisgveneri**
## PR Type

[ ] Bugfix
[ x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[ ] Build-related changes
[ ] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: 

## What Is the Current Behavior?

Closes #948 

Quote products cannot be removed from the cart once they have been added to it.

## What Is the New Behavior?

Quote products can now be removed from the cart.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[ x] No

## ToDo
[ ] Unit tests
[ ] Add localization key 'quote.basket_items.label' for all languages (now is only for US)

### CODE REVIEW

File: src/app/extensions/quoting/facades/quoting.facade.ts [hunk 1]
```diff
@@ -58,6 +58,13 @@ export class QuotingFacade {
     );
   }
```
Comment (dhhyi) [inline src/app/extensions/quoting/facades/quoting.facade.ts:59]: above: please move `this.store.dispatch(loadQuoting());` into a separate facade method `loadQuoting()` and reuse it in the effect above and in the new `quoting-basket-line-items.component`, that way it is not that implicitly used.

File: src/app/extensions/quoting/shared/quoting-basket-line-items/quoting-basket-line-items.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      map(basket => Object.entries(groupBy(basket.lineItems, 'quote')))
+    );
+  }
+
+  getName(quoteId: string) {
+    return this.quotingFacade.name$(quoteId);
+  }
+
+  onDeleteQuote(lineItems: LineItem[]) {
+    lineItems.forEach(item => this.checkoutFacade.deleteBasketItem(item.id));
```
Comment (dhhyi) [inline src/app/extensions/quoting/shared/quoting-basket-line-items/quoting-basket-line-items.component.ts:34]: It looks like we shouldn't use multiple deletes here, as the first delete deletes all quote items from the basket already, and the subsequent deletes fail.
@SGrueber Is there a better way to delete quotes from the basket that via items? I did not really find one in the API doc.

File: src/app/extensions/quoting/shared/quoting-basket-line-items/quoting-basket-line-items.component.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<ng-container *ngFor="let item of lineItems$ | async">
+  <div class="row">
+    <div class="col-md-6 my-auto">
+      <h1 *ngIf="item[0] !== 'undefined'">{{ getName(item[0]) | async }}</h1>
+    </div>
```
Comment (dhhyi) [inline src/app/extensions/quoting/shared/quoting-basket-line-items/quoting-basket-line-items.component.html:5]: - The ngIf should probably be on the div with the 'row', so it doesn't apply for normal items
- Please introduce a translation key `From Quote {{quoteId}}`
- I don't think the column layout is set up correctly. On smaller devices, the recycle bin icon breaks in a wired way to the next line.

**Review by dhhyi — CHANGES_REQUESTED** [review]

File: src/app/extensions/quoting/shared/quoting-basket-line-items/quoting-basket-line-items.component.ts (hunk 2 above)
Comment (dhhyi) [reply src/app/extensions/quoting/shared/quoting-basket-line-items/quoting-basket-line-items.component.ts:34]: Maybe for now it is best to just delete the first line item.

Comment (SGrueber) [reply src/app/extensions/quoting/shared/quoting-basket-line-items/quoting-basket-line-items.component.ts]: This is the way to go. If a basket item is deleted that is based on a quote all other basket items of this quote will also be deleted by this REST call.

**Review by dhhyi — APPROVED** [review]

**Review by SGrueber — APPROVED** [review]

**Pull request merged**